test/test_sparse.py

```diff
@@ -1694,6 +1694,129 @@ def test_log1p(self):
             device=self.device)
         self._test_log1p_tensor(input, torch.zeros([5, 6, 0]))
 
+    def _test_neg_negative(self, sparse_tensor):
+        dense_tensor = sparse_tensor.to_dense()
+        expected_output = dense_tensor.neg()
+        self.assertEqual(expected_output, torch.neg(sparse_tensor).to_dense())
```

Comment:
```
for op in (torch.neg, torch.negative):
  ....
```

should save you some redundancy in these blocks.

I notice that only the "out" variant is tested. This goes back to my dispatch question, is there a problem with just calling torch.neg(t) on a sparse tensor t?

Reply:
> I notice that only the "out" variant is tested. This goes back to my dispatch question, is there a problem with just calling torch.neg(t) on a sparse tensor t?

No, there is no problem with calling `torch.neg(t)` on a sparse tensor t. I think the tests check all versions: `torch.neg(t)`,  `torch.neg(sparse_tensor, out=sparse_tensor_out)` and `sparse_tensor.neg()`. May be I'm missing something ?

Reply:
My mistake, I thought only the out branch was tested but you're correct, of course.